google/src/main/scala/org/broadinstitute/dsde/rawls/google/GoogleUtilities.scala

```diff
@@ -36,11 +36,22 @@ trait GoogleUtilities extends LazyLogging with Retry {
     }
   }
 
-  protected def retryWhen500orGoogleError[T](op: () => T) = {
+  protected def whenGoogle404: Predicate[Throwable] = throwable => {
+    throwable match {
+      case t: GoogleJsonResponseException if t.getStatusCode == 404 => true
+      case _ => false
+    }
+  }
+
+  protected def retryWhen[T](pred: Predicate[Throwable])(op: () => T): Future[T] = {
```

Comment:
I feel like all these `op: () => T` (which is a Function1[Unit, T]) could be written using by-name parameters instead: `op: => T`. But I'm not sure if there is a good reason for this, so I left it as is.